bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,197 @@
+                            if x_ != -1 and x_ != 10 and y_ != -1 and y_ != 10 and board[y_][x_] == "bomb":
+                                bombs += 1
+
+                    board[y][x] = bombs
+        return board
+
+    @staticmethod
+    def format_for_discord(board: GameBoard) -> str:
+        """Format the board to a string for discord."""
+        mapping = {
+            0: ":stop_button:",
+            1: ":one:",
+            2: ":two:",
+            3: ":three:",
+            4: ":four:",
+            5: ":five:",
+            6: ":six:",
+            7: ":seven:",
+            8: ":eight:",
+            9: ":nine:",
+            10: ":keycap_ten:",
+            "bomb": ":bomb:",
+            "hidden": ":grey_question:",
+            "flag": ":triangular_flag_on_post:"
+        }
+
+        discord_msg = (
+            ":stop_button:    :regional_indicator_a::regional_indicator_b::regional_indicator_c:"
+            ":regional_indicator_d::regional_indicator_e::regional_indicator_f::regional_indicator_g:"
+            ":regional_indicator_h::regional_indicator_i::regional_indicator_j:\n\n"
+        )
+        rows: typing.List[str] = []
+        for row_number, row in enumerate(board):
+            new_row = mapping[row_number + 1] + "    "
+            for cell in row:
+                new_row += mapping[cell]
+            rows.append(new_row)
+
+        discord_msg += "\n".join(rows)
+        return discord_msg
+
+    @commands.command(name="minesweeper")
+    async def minesweeper_command(self, ctx: commands.Context, bomb_chance: float = .2) -> None:
+        """Start a game of minesweeper."""
+        if ctx.author.id in self.games.keys():  # Player is already playing
+            msg = await ctx.send(f"{ctx.author.mention} you already have a game running")
+            await msg.delete(delay=2)
+            await ctx.message.delete(delay=2)
+            return
+
+        # Add game to list
+        board: GameBoard = self.generate_board(bomb_chance)
+        revealed_board: GameBoard = [["hidden" for _ in range(10)] for _ in range(10)]
+
+        await ctx.send(f"{ctx.author.mention} is playing minesweeper")
+        chat_msg = await ctx.send(self.format_for_discord(revealed_board))
+
+        await ctx.author.send("play by typing: `.reveal xy [xy]` or `.flag xy [xy]` \n"
+                              "close the game with `.end`\n"
+                              "cords must be in format `<letter><number>`")
+        dm_msg = await ctx.author.send(self.format_for_discord(revealed_board))
+
+        self.games[ctx.author.id] = {
+            "board": board,
+            "revealed": revealed_board,
+            "dm_msg": dm_msg,
+            "chat_msg": chat_msg
+        }
+
+    @staticmethod
+    def get_cords(value1: str, value2: str) -> typing.Tuple[int, int]:
+        """Take in 2 values for the cords and turn them into numbers"""
+        return ord(value1.lower()) - 97, int(value2) - 1
```

Comment:
This throws an error if the users uses <num><letter>.  I believe this should be caught and either try to flip them, or give the user a message that it was the wrong format and what is correct.